### PR TITLE
[mesheryctl] Add connection view subcommand

### LINKED ISSUE #10965: [mesheryctl] New command exp connection view
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

### Current Behavior

We have the following `mesheryctl exp connections` subcommands available
- list
- delete

### Desired Behavior

We want to add  `view` subcommand to display the details of a connection based on his id

### Mockups

---

### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by singh1203**
**Notes for Reviewers**
This PR depends on PR #11478 since it requires the connectionbyId handler function to work as expected.

This PR fixes #10965

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 397ecb0805a3b2dedfe689dde439d136a599212e at: https://66a8ec1cc2b33e1c12b965f3--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 58381528129748161aed48797e7e213b1b2c3080 at: https://66a9ac690e70be4483690499--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 30791632ba7c1a44f0175d0b52c520d26c033434 at: https://66a9adc47dbfe448f2fb42a5--meshery-docs-preview.netlify.app

**Comment by Ashparshp** [comment]
@singh1203 Let's discuss this issue during the Meshery dev meeting call. Please add it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?usp=drivesdk).

**Comment by leecalcote** [comment]
@lekaf974 eyes on this would be great.

File: docs/pages/reference/mesheryctl/mesheryctl-exp-connections-view.md
Comment (alphaX86) [inline docs/pages/reference/mesheryctl/mesheryctl-exp-connections-view.md:1]: This page would anyway get added by the workflow...

File: mesheryctl/internal/cli/root/connections/list.go [hunk 1]
```diff
@@ -70,27 +70,25 @@ mesheryctl exp connections list --page 2
 		if cmd.Flags().Changed("page") {
 			url = fmt.Sprintf("%s/api/integrations/connections?page=%d", baseUrl, pageNumberFlag)
 		} else {
-			url = fmt.Sprintf("%s/api/integrations/connections?pagesize=all", baseUrl)
+			url = fmt.Sprintf("%s/api/integrations/connections", baseUrl)
```
Comment (alphaX86) [inline mesheryctl/internal/cli/root/connections/list.go:73]: Is the API call correct for this one @singh1203 ?

File: mesheryctl/internal/cli/root/connections/view.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"io"
+	"net/http"
+)
+
+var viewConnectionCmd = &cobra.Command{
+	Use:   "view",
+	Short: "View a connection by id",
+	Long: `View a specific connection details by id
```
Comment (alphaX86) [inline mesheryctl/internal/cli/root/connections/view.go:35]:
```suggestion
	Short: "View a connection by ID",
	Long: `View a specific connection details by ID
```

File: mesheryctl/internal/cli/root/connections/view.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+Display the details of a specific connection based on id provided`,
```
Comment (alphaX86) [inline mesheryctl/internal/cli/root/connections/view.go:37]:
```suggestion
Display the details of a specific connection based on ID provided`,
```

File: docs/_data/mesheryctlcommands/cmds.yml [hunk 4]
```diff
@@ -1020,6 +1020,10 @@ exp:
           count:
             name: --count
             description: (optional) Get the number of exp connections in total
+      view:
+        name: view
+        description: View a specific connection details by id
```
Comment (alphaX86) [inline docs/_data/mesheryctlcommands/cmds.yml:1025]:
```suggestion
        description: View a specific connection details by ID
```

**Comment by l5io** [comment]
🚀 Preview for commit 16d4dcc97a53100c3ea554f782277f759c5fd70b at: https://66af9a200e15abae65459c6d--meshery-docs-preview.netlify.app

File: mesheryctl/internal/cli/root/connections/list.go (hunk 1 above)
Comment (singh1203) [reply mesheryctl/internal/cli/root/connections/list.go]: Apologies for wrong changes here.

**Comment by l5io** [comment]
🚀 Preview for commit 377c07105cedff5f1e460f6f1ffafc4343f96dc7 at: https://66af9d20fe7aadc25221834a--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 8ec93564511df1fb2b6d4145b2dc2c07a5b9d142 at: https://66b2befa23256e272fb49583--meshery-docs-preview.netlify.app

**Comment by Ashparshp** [comment]
@singh1203 let's discuss this PR during the Meshery dev meeting call, if you would. Please add it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?usp=drivesdk).

**Comment by l5io** [comment]
🚀 Preview for commit d9bd42463b955d9c194bc0f7ac8b6888282162b0 at: https://66b589eb586e72dbaaf35297--meshery-docs-preview.netlify.app

**Review by alphaX86 — COMMENTED** [review]
Is this subcommand fully tested @singh1203 ?

**Comment by alphaX86** [comment]
> > Is this subcommand fully tested @singh1203 ?
> 
> No, we can't fully test this subcommand until we have the connection by ID handler implementation. Also, running the meshery server in a local environment doesn't allow us to test the mesheryctl binary change. Do you have any guidance on this?

We can run the modified `mesheryctl` binary by building a new one in local. That is, run `make` in the `mesheryctl` folder of repo. Then you'll get a generated binary. Run with that binary using `./mesheryctl` or `./mesheryctl.exe`

**Comment by l5io** [comment]
🚀 Preview for commit b381a109747a33b1c76c1b7d5eb8c5271d1424c3 at: https://66b8d46cf205a46c2d4fe202--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 31bc4f0574698a6c9ab8ffc29f6e3937b88f2ab7 at: https://66bdf7d9fb8e5a1f48d20d72--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit ceb77ccc261129353317b77e4d9d3a3492d74b27 at: https://66c5eb55f8ce5234334eab44--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit a001bd678e39738547d56aa8c86c44784faa7476 at: https://66e3c81fc19b06a8c578ec70--meshery-docs-preview.netlify.app

**Pull request closed without merging**